### PR TITLE
Update documentation for Paket 3 and fix spelling and grammar mistakes

### PR DESCRIPTION
**Pull request by Kazark**
+ Remove references to alpha/prerelease versions of Paket 3 since Paket 3 has been released
+ Mention Git as a new dependency type in the "Sources" section of the dependencies file documentation
+ Find and fix numerous spelling and casing mistakes using spell checker in Vim
+ Fix a comma splice by breaking it into two sentences

### CODE REVIEW

File: docs/content/git-dependencies.md [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 Paket allows you to automatically manage the linking of files from any git repo.
 
-<blockquote>This feature is only available in Paket 3.0 prereleases.</blockquote>
+<blockquote>This feature is only available in Paket 3.0.</blockquote>
```
Comment (forki) [inline docs/content/git-dependencies.md]: I think we can remvoe the full paragraph

**Comment by forki** [comment]
:heart: Awesome. Thanks

File: docs/content/dependencies-file.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ Only direct dependencies should be listed and you can use the [`paket simplify`
 Paket supports the following source types:
 
 * [NuGet](nuget-dependencies.html)
+* [Git](git-dependencies.html)
```
Comment (Kazark) [inline docs/content/dependencies-file.md]: Uh-oh, looks like this was a goof-up

Comment (forki) [inline docs/content/dependencies-file.md]: just send another PR ;-) thanks

Comment (Kazark) [inline docs/content/dependencies-file.md]: Actually the diff was confusing me. Double-checked both in my source and in the updated online docs and we're good. :100:
